src/rules/convertRules.test.ts

```diff
@@ -183,6 +184,51 @@ describe("convertRules", () => {
         );
     });
 
+    it("merges and deduplicates rule notices", () => {
```

Comment:
This test is mostly copy pasta. I don't think it has anything unnecessary but happy to remove stuff if I missed something :)

Reply:
Yeah, I think this file has a lot of [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)/[WET](https://dev.to/wuz/stop-trying-to-be-so-dry-instead-write-everything-twice-wet-5g33) violations. Either the source function is doing too much and/or the unit tests should have common logic moved into helpers.

A few options for you here, all of which I have no issue with:
* Ignore this comment and continue eating the 🍝
* File a followup issue to investigate this
* In this PR, move shared logic from these unit tests into a function in this file
* In this PR, refactor some logic out of `convertRules` so edge cases can be directly unit tested - for example, the contents of the `for` loop.

Reply:
Gonna ignore for now and create a follow-up issue.